ga4gh/datamodel/genotype_phenotype.py

```diff
@@ -0,0 +1,624 @@
+            association['sources']
+            )
+        evidence = protocol.Evidence()
+        phenotype = association['phenotype']
+
+        term = protocol.OntologyTerm()
+        term.term = association['evidence_type']
+        term.id = phenotype['id']
+        term.source_version = self._version
+        term.source_name = self._getPrefix(
+            self._getPrefixURL(association['id']))
+        evidence.evidence_type.MergeFrom(term)
+
+        evidence.description = self._getIdentifier(association['evidence'])
+        # Store publications list of sources
+        for source in association['sources'].split("|"):
+            evidence.info['publications'].values.add().string_value = source
+        fpa.evidence.extend([evidence])
+
+        # map environment (drug)
+        environmentalContext = protocol.EnvironmentalContext()
+        environment = association['environment']
+        environmentalContext.id = environment['id']
+        environmentalContext.description = association['environment_label']
+
+        term = protocol.OntologyTerm()
+        term.term = environment['id']
+        term.id = 'http://purl.obolibrary.org/obo/RO_0002606'
+        term.source_version = self._version
+        term.source_name = self._getPrefix(
+            self._getPrefixURL(association['id']))
+        environmentalContext.environment_type.MergeFrom(term)
+
+        fpa.environmental_contexts.extend([environmentalContext])
+
+        phenotypeInstance = protocol.PhenotypeInstance()
+        term = protocol.OntologyTerm()
+        term.term = phenotype[TYPE]
+        term.id = phenotype['id']
+        term.source_version = self._version
+        term.source_name = self._getPrefix(
+            self._getPrefixURL(association['id']))
+        phenotypeInstance.type.MergeFrom(term)
+
+        phenotypeInstance.description = phenotype[LABEL]
+        phenotypeInstance.id = phenotype['id']
+        fpa.phenotype.MergeFrom(phenotypeInstance)
+        fpa.phenotype_association_set_id = self.getId()
+        return fpa
+
+
+class RdfPhenotypeAssociationSet(G2PUtility, AbstractPhenotypeAssociationSet):
+    """
+    An rdf object store.  The cancer genome database
+    [Clinical Genomics Knowledge Base]
+    (http://nif-crawler.neuinfo.org/monarch/ttl/cgd.ttl),
+    published by the Monarch project, was the source of Evidence.
+    """
+    def __init__(self, parentContainer, localId, dataDir):
+        super(RdfPhenotypeAssociationSet, self).__init__(
+            parentContainer, localId)
+        """
+        Initialize dataset, using the passed dict of sources
+        [{source,format}] see rdflib.parse() for more
+        If path is set, this backend will load itself
+        """
+
+        # initialize graph
+        self._rdfGraph = rdflib.ConjunctiveGraph()
+        # save the path
+        self._dataUrl = dataDir
+
+        try:
+            self._scanDataFiles(dataDir, ['*.ttl', '*.xml'])
+        except AttributeError:
```

Comment:
This looks really dangerous... This will mask any programming errors in the _scanDataFiles method that look for an unknown attribute. Why is this here anyway? The user is much better off if errors happen early and the server fails noisily.